src/test/java/mate/academy/dao/impl/AbstractTest.java

```diff
@@ -0,0 +1,142 @@
+package mate.academy.dao.impl;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+import javax.sql.DataSource;
+import org.hibernate.Interceptor;
+import org.hibernate.SessionFactory;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.Configuration;
+import org.hsqldb.jdbc.JDBCDataSource;
+import org.junit.Before;
+
+public abstract class AbstractTest {
+    protected interface DataSourceProvider {
+
+        enum IdentifierStrategy {
+            IDENTITY,
+            SEQUENCE
+        }
+
+        enum Database {
+            HSQLDB,
+        }
+
+        String hibernateDialect();
+
+        DataSource dataSource();
+
+        Class<? extends DataSource> dataSourceClassName();
+
+        Properties dataSourceProperties();
+
+        List<IdentifierStrategy> identifierStrategies();
+
+        Database database();
+    }
+
+    private SessionFactory factory;
+
+    @Before
+    public void init() {
+        factory = newSessionFactory();
+    }
+
+
+    private SessionFactory newSessionFactory() {
+        Properties properties = getProperties();
+        Configuration configuration = new Configuration().addProperties(properties);
+        for(Class<?> entityClass : entities()) {
```

Comment:
it will be nice to run this code through checkstyle plugin
```suggestion
        for (Class<?> entityClass : entities()) {
```